# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1522623396 -7200
#      Mon Apr 02 00:56:36 2018 +0200
# Branch stable
# Node ID f50fa34827fb56184734749e3c618f8666e85dc1
# Parent  08d19ea507424f7f90d981c085f55a641939934f
use spin boxes for col/row/ width/height in settings for variable editor

* settings-dialog.cc (ctor): set values in the spin boxes from settings file;
  (write_changed_settings): store spin box values to settings file

* settings-dialog.ui: replace test labels by spin boxes for column width
  and row height of variable editor

diff --git a/libgui/src/settings-dialog.cc b/libgui/src/settings-dialog.cc
--- a/libgui/src/settings-dialog.cc
+++ b/libgui/src/settings-dialog.cc
@@ -563,20 +563,20 @@ settings_dialog::settings_dialog (QWidge
   read_workspace_colors (settings);
   // hide tool tips
   ui->cb_hide_tool_tips->setChecked (settings->value ("workspaceview/hide_tool_tips", false).toBool ());
 
   // terminal colors
   read_terminal_colors (settings);
 
   // variable editor
-  ui->varedit_columnWidth->setText (settings->value ("variable_editor/column_width", "100").toString ());
+  ui->varedit_columnWidth->setValue (settings->value ("variable_editor/column_width", 100).toInt ());
   ui->varedit_autoFitColumnWidth->setChecked (settings->value ("variable_editor/autofit_column_width", false).toBool ());
   ui->varedit_autofitType->setCurrentIndex (settings->value ("autofit_type", 0).toInt ());
-  ui->varedit_rowHeight->setText (settings->value ("variable_editor/row_height", "2").toString ());
+  ui->varedit_rowHeight->setValue (settings->value ("variable_editor/row_height", 10).toInt ());
   ui->varedit_rowAutofit->setChecked (settings->value ("variable_editor/autofit_row_height", true).toBool ());
 
   ui->varedit_font->setCurrentFont (QFont (settings->value ("variable_editor/font_name", settings->value ("terminal/fontName", "Courier New")).toString ()));
   ui->varedit_fontSize->setValue (settings->value ("variable_editor/font_size", QVariant (10)).toInt ());
   connect (ui->varedit_useTerminalFont, SIGNAL (toggled (bool)),
            ui->varedit_font, SLOT (setDisabled (bool)));
   connect (ui->varedit_useTerminalFont, SIGNAL (toggled (bool)),
            ui->varedit_fontSize, SLOT (setDisabled (bool)));
@@ -956,18 +956,18 @@ settings_dialog::write_changed_settings 
   settings->setValue ("workspaceview/hide_tool_tips", ui->cb_hide_tool_tips->isChecked ());
 
   // Terminal
   write_terminal_colors (settings);
 
   // Variable editor
   settings->setValue ("variable_editor/autofit_column_width", ui->varedit_autoFitColumnWidth->isChecked ());
   settings->setValue ("variable_editor/autofit_type", ui->varedit_autofitType->currentIndex ());
-  settings->setValue ("variable_editor/column_width", ui->varedit_columnWidth->text ());
-  settings->setValue ("variable_editor/row_height", ui->varedit_rowHeight->text ());
+  settings->setValue ("variable_editor/column_width", ui->varedit_columnWidth->value ());
+  settings->setValue ("variable_editor/row_height", ui->varedit_rowHeight->value ());
   settings->setValue ("variable_editor/autofit_row_height", ui->varedit_rowAutofit->isChecked ());
   settings->setValue ("variable_editor/use_terminal_font", ui->varedit_useTerminalFont->isChecked ());
   settings->setValue ("variable_editor/alternate_rows", ui->varedit_alternate->isChecked ());
   settings->setValue ("variable_editor/font_name", ui->varedit_font->currentFont ().family ());
   settings->setValue ("variable_editor/font_size", ui->varedit_fontSize->value ());
   write_varedit_colors (settings);
 
   // shortcuts
diff --git a/libgui/src/settings-dialog.ui b/libgui/src/settings-dialog.ui
--- a/libgui/src/settings-dialog.ui
+++ b/libgui/src/settings-dialog.ui
@@ -27,17 +27,17 @@
     <widget class="QTabWidget" name="tabWidget">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="currentIndex">
-      <number>1</number>
+      <number>8</number>
      </property>
      <widget class="QWidget" name="tab_general">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <attribute name="title">
        <string>General</string>
       </attribute>
@@ -504,17 +504,17 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>642</width>
+            <width>645</width>
             <height>1012</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_16">
            <item>
             <widget class="QGroupBox" name="groupBox_5">
              <property name="title">
               <string>General</string>
@@ -1818,18 +1818,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_4">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>481</width>
-            <height>230</height>
+            <width>658</width>
+            <height>573</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <layout class="QGridLayout" name="gridLayout_7">
                <property name="sizeHint" stdset="0">
@@ -2099,18 +2099,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_5">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>474</width>
-            <height>196</height>
+            <width>658</width>
+            <height>573</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_8">
            <item row="0" column="0">
             <widget class="QGroupBox" name="groupBox_4">
              <property name="title">
               <string>Behavior</string>
              </property>
@@ -2243,18 +2243,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_6">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>200</width>
-            <height>77</height>
+            <width>658</width>
+            <height>573</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_19">
            <item>
             <layout class="QGridLayout" name="gridLayout_14">
              <property name="topMargin">
               <number>0</number>
              </property>
@@ -2312,18 +2312,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_8">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>364</width>
-            <height>210</height>
+            <width>658</width>
+            <height>573</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_25">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_26">
              <property name="sizeConstraint">
               <enum>QLayout::SetDefaultConstraint</enum>
              </property>
@@ -2511,18 +2511,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_7">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>529</width>
-            <height>199</height>
+            <width>658</width>
+            <height>573</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_20">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <widget class="QCheckBox" name="checkbox_allow_web_connect">
                <property name="text">
@@ -2670,33 +2670,53 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_3">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>615</width>
-            <height>169</height>
+            <width>658</width>
+            <height>573</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_vesc_manual">
            <property name="geometry" stdset="0">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>678</width>
              <height>384</height>
             </rect>
            </property>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_ve">
              <item>
               <layout class="QGridLayout" name="gridLayout_ve">
+               <item row="0" column="3">
+                <widget class="QLabel" name="label_26">
+                 <property name="text">
+                  <string>Font size</string>
+                 </property>
+                 <property name="alignment">
+                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
+                 </property>
+                </widget>
+               </item>
+               <item row="2" column="2" colspan="2">
+                <widget class="QCheckBox" name="varedit_rowAutofit">
+                 <property name="text">
+                  <string>Plus font height</string>
+                 </property>
+                 <property name="checked">
+                  <bool>true</bool>
+                 </property>
+                </widget>
+               </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="label_27">
                  <property name="text">
                   <string>Default row height</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
@@ -2704,23 +2724,16 @@
                  <property name="text">
                   <string>Font</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
-               <item row="1" column="1">
-                <widget class="QLineEdit" name="varedit_columnWidth">
-                 <property name="text">
-                  <string>20</string>
-                 </property>
-                </widget>
-               </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_ve_colwidth">
                  <property name="text">
                   <string>Default column width</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
@@ -2738,59 +2751,32 @@
                </item>
                <item row="0" column="4">
                 <widget class="QSpinBox" name="varedit_fontSize">
                  <property name="value">
                   <number>10</number>
                  </property>
                 </widget>
                </item>
-               <item row="0" column="3">
-                <widget class="QLabel" name="label_26">
-                 <property name="text">
-                  <string>Font size</string>
-                 </property>
-                 <property name="alignment">
-                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
-                 </property>
-                </widget>
-               </item>
-               <item row="2" column="1">
-                <widget class="QLineEdit" name="varedit_rowHeight">
-                 <property name="text">
-                  <string>2</string>
-                 </property>
-                </widget>
-               </item>
                <item row="0" column="1" colspan="2">
                 <widget class="QFontComboBox" name="varedit_font">
                  <property name="currentFont">
                   <font>
                    <family>Liberation Mono</family>
                   </font>
                  </property>
                 </widget>
                </item>
                <item row="1" column="2">
                 <widget class="QCheckBox" name="varedit_autoFitColumnWidth">
                  <property name="text">
                   <string>Autofit</string>
                  </property>
                 </widget>
                </item>
-               <item row="2" column="2" colspan="2">
-                <widget class="QCheckBox" name="varedit_rowAutofit">
-                 <property name="text">
-                  <string>Plus font height</string>
-                 </property>
-                 <property name="checked">
-                  <bool>true</bool>
-                 </property>
-                </widget>
-               </item>
                <item row="1" column="3" colspan="2">
                 <widget class="QComboBox" name="varedit_autofitType">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="currentIndex">
                   <number>0</number>
                  </property>
@@ -2814,16 +2800,33 @@
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
+               <item row="2" column="1">
+                <widget class="QSpinBox" name="varedit_rowHeight">
+                 <property name="value">
+                  <number>10</number>
+                 </property>
+                </widget>
+               </item>
+               <item row="1" column="1">
+                <widget class="QSpinBox" name="varedit_columnWidth">
+                 <property name="maximum">
+                  <number>500</number>
+                 </property>
+                 <property name="value">
+                  <number>100</number>
+                 </property>
+                </widget>
+               </item>
               </layout>
              </item>
              <item>
               <widget class="QGroupBox" name="varedit_colors_box">
                <property name="title">
                 <string>Variable Editor Colors</string>
                </property>
               </widget>
